tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -6,100 +6,105 @@
 from openvino.tools.mo.front.extractor import raise_no_node, raise_node_name_collision
 from openvino.tools.mo.utils.error import Error
 
-from openvino.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from openvino.frontend import (
+    InputModel,
+)  # pylint: disable=no-name-in-module,import-error
 
 import numpy as np
 
 from enum import Enum
 
-class MatchType(Enum):
-    PRE = 1
-    POST = 2
 
+class IOType(Enum):
+    Input = 1
+    Output = 2
 
-def decode_name_with_port(input_model: InputModel, node_name: str, framework=""):
+
+def decode_name_with_port(
+    input_model: InputModel, node_name: str, framework="", io_type=IOType.Input
```

Comment:
You can provide types along with default values, as well as a return type: 
`def foo(opts: dict = {}) -> float:`

Reply:
Done